sklearn/metrics/_ranking.py

```diff
@@ -903,6 +904,15 @@ def precision_recall_curve(y_true, probas_pred, *, pos_label=None, sample_weight
     return np.hstack((precision[sl], 1)), np.hstack((recall[sl], 0)), thresholds[sl]
 
 
+@validate_params(
+    {
+        "y_true": ["array-like", "sparse matrix"],
```

Comment:
A sparse matrix does not seem to be valid. Since a sparse matrix is 2D, the type of target is misinterpreted resulting in an error.
```suggestion
        "y_true": ["array-like"],
```